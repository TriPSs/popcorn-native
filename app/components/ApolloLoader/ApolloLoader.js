import React from 'react'
import { ApolloProvider } from '@apollo/react-common'

import getClient from 'modules/GraphQL/Apollo'

export default class ApolloLoader extends React.PureComponent {

  state = {
    loading: true,
  }

  client = null

  async componentDidMount() {
    const { host } = this.props

    this.client = await getClient(host)

    this.setState({
      loading: false,
    })
  }

  render() {
    const { children } = this.props
    const { loading } = this.state

    if (loading) {
      return null
    }

    return (
      <ApolloProvider client={this.client}>
        {children}
      </ApolloProvider>
    )
  }

}
