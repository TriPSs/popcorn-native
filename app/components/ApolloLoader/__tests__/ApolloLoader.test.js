import React from 'react'
import { View } from 'react-native'
import { shallow } from 'enzyme'

import getClient from 'modules/GraphQL/Apollo'

import ApolloLoader from '../ApolloLoader'

jest.mock('modules/GraphQL/Apollo', () => jest.fn())

describe('(Component) ApolloClient', () => {

  const component = (
    <ApolloLoader>
      <View />
    </ApolloLoader>
  )

  it('should start loading', () => {
    const wrapper = shallow(component)

    expect(wrapper).toMatchSnapshot()
  })

  it('should finish loading', async() => {
    const wrapper = shallow(component)

    await wrapper.instance().componentDidMount()
    expect(await getClient).toHaveBeenCalled()
    expect(wrapper).toMatchSnapshot()
  })
})
