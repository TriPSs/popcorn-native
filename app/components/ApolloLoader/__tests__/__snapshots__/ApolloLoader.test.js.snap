// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ApolloClient should finish loading 1`] = `
<ApolloProvider>
  <View />
</ApolloProvider>
`;

exports[`(Component) ApolloClient should start loading 1`] = `""`;
